flocker/node/agents/test/test_blockdevice.py

```diff
@@ -146,6 +154,235 @@ def test_only_unattached_devices(self):
         self.assertDiscoveredState(self.deployer, [])
 
 
+class BlockDeviceDeployerDestructionCalculateNecessaryStateChangesTests(
+        SynchronousTestCase
+):
+    """
+    Tests for ``BlockDeviceDeployer.calculate_necessary_state_changes``
+    in the cases relating to dataset destruction.
+    """
+    def test_undeleted_dataset_not_deleted(self):
+        """
+        ``BlockDeviceDeployer.calculate_necessary_state_changes`` does not
+        calculate a change to destroy datasets that are not marked as deleted
+        in the configuration.
+        """
+        dataset_id = uuid4()
```

Comment:
Looks like some of this repeats over tests? Can it be moved into utility method?